src/js/store/factory/marks/Text.ts

```diff
@@ -40,57 +42,30 @@ function Text() {
  * @param {string} text.type - A mark type, presumably "text"
  * @returns {Object} A dictionary of stream definitions keyed by signal name
  */
-Text.getHandleStreams = function(text) {
-  const sg = require('../../../ctrl/signals'),
-      at = anchorTarget.bind(null, text, 'handles'),
-      id = text._id,
-      x = propSg(id, 'text', 'x'),
-      y = propSg(id, 'text', 'y'),
-      fontSize = propSg(id, 'text', 'fontSize'),
-      DELTA = sg.DELTA,
-      DX = DELTA + '.x',
-      DY = DELTA + '.y',
-      streams = {};
+export function getHandleStreams(text: TextRecord): {[s: string]: OnEvent[];} {
+  const sg = require('../../../ctrl/signals');
+  const at = anchorTarget.bind(null, text, 'handles');
+  const id = text._id;
+  const x = propSg(id, 'text', 'x');
+  const y = propSg(id, 'text', 'y');
+  const fontSize = propSg(id, 'text', 'fontSize');
+  const DELTA = sg.DELTA;
+  const DX = DELTA + '.x';
+  const DY = DELTA + '.y';
+  const streams: {[s: string]: OnEvent[];} = {};
 
   streams[x] = [{
-    type: DELTA, expr: test(at(), x + '+' + DX, x)
+    events: DELTA, update: test(at(), x + '+' + DX, x)
   }];
   streams[y] = [{
-    type: DELTA, expr: test(at(), y + '+' + DY, y)
+    events: DELTA, update: test(at(), y + '+' + DY, y)
   }];
   // Allow upper-left and lower-right handles to control font size
   streams[fontSize] = [
-    {type: DELTA, expr: test(at('left') + '&&' + at('top'), fontSize + '-' + DX, fontSize)},
-    {type: DELTA, expr: test(at('right') + '&&' + at('bottom'), fontSize + '+' + DX, fontSize)},
-    {type: DELTA, expr: test(at('left') + '&&' + at('top'), fontSize + '-' + DY, fontSize)},
-    {type: DELTA, expr: test(at('right') + '&&' + at('bottom'), fontSize + '+' + DY, fontSize)}
+    {events: DELTA, update: test(at('left') + '&&' + at('top'), fontSize + '-' + DX, fontSize)},
+    {events: DELTA, update: test(at('right') + '&&' + at('bottom'), fontSize + '+' + DX, fontSize)},
+    {events: DELTA, update: test(at('left') + '&&' + at('top'), fontSize + '-' + DY, fontSize)},
+    {events: DELTA, update: test(at('right') + '&&' + at('bottom'), fontSize + '+' + DY, fontSize)}
   ];
   return streams;
 };
-
-/**
- * @property alignments {string[]} Valid align properties for vega text marks
- */
-Text.alignments = ['left', 'center', 'right'];
```

Comment:
I'm curious why we're deleting all of these? I think we used these to populate drop downs in the UI. Are we getting these values some other way (e.g., maybe directly from the typings somehow?).

Reply:
thank you for catching this, it was an overzealous delete. i added these back with vega types here: https://github.com/vega/lyra/pull/427/commits/6d9c1aca476bf67113b7cc1ae5308c88937f45c2